typedb/stream/response_part_iterator.py

```diff
@@ -78,6 +77,7 @@ def _has_next(self) -> bool:
 
     def __next__(self) -> transaction_proto.Transaction.ResPart:
         if not self._has_next():
+            self._bidirectional_stream.done(self._request_id)
```

Comment:
we also let the transaction stream know this query stream is done so it can be removed